ui/src/app/components/copy-snippet-modal.tsx

```diff
@@ -0,0 +1,46 @@
+import {faCheck, faClipboard} from '@fortawesome/free-solid-svg-icons';
+import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
+import {Button} from 'app/components/buttons';
+import {Modal, ModalBody, ModalFooter, ModalTitle} from 'app/components/modals';
+import colors from 'app/styles/colors';
+import {useState} from 'react';
+import * as React from 'react';
+
+interface Props {
+  title: string;
+  copyText: string;
+  closeFunction: Function;
+}
+
+export const CopySnippetModal = ({title, copyText, closeFunction}: Props) => {
+  const [isCopied, setIsCopied] = useState(false);
+  const [timer, setTimer] = useState(null);
+
+  return (
+    <Modal>
+      <ModalTitle>{title}</ModalTitle>
+      <ModalBody style={{color: colors.primary}}>{copyText}</ModalBody>
+      <ModalFooter style={{justifyContent: 'space-between'}}>
+        <Button type='secondaryLight'
+                style={{paddingLeft: '0'}}
+                onClick={() => {
+                  // @ts-ignore: Unreachable code error. TODO: We can remove this once TS is upgraded to >= 3.4
```

Comment:
huh. I'm curious why it thinks this is unreachable, do you know offhand?

Reply:
that was just an auto generated error from IntelliJ which doesn't make sense now that I think about it. I changed the message. good catch!

Reply:
... and I accidentally merged before uploading my commit. I'll throw it in with my next PR